sky/resources.py

```diff
@@ -2219,3 +2318,96 @@ def _maybe_add_docker_prefix_to_image_id(
     for k, v in image_id_dict.items():
         if not v.startswith('docker:'):
             image_id_dict[k] = f'docker:{v}'
+
+
+def _get_devices(memory: float,
+                 plus: bool = False,
+                 manufacturer: Optional[str] = None) -> List[str]:
+    """Returns a list of device names that meet the memory and manufacturer
+       requirements.
+
+    Args:
+        memory: The minimum memory size in GB.
+        plus: If True, returns devices with memory >= memory, otherwise returns
+              devices with memory == memory.
+        manufacturer: The manufacturer of the GPU.
+    """
+    common = CATALOG_DIR + '/' +\
+             CATALOG_SCHEMA_VERSION + '/' +\
+             ACCELERATORS_PATH
+
+    # Read the accelerators CSV file
+    df = pd.read_csv(common)
```

Comment:
We should avoid having some custom logic outside the service_catalog module that's reading from the catalog. The catalog is lazy fetched the first time you start skypilot, so we can't guarantee it exists here. We should be calling into code in sky.catalog or moving this code into there, and using the primitives there for accessing the catalog. 
Specifically we need to use sky.catalog.common.read_catalog to dynamically pull from the service catalog, otherwise the file we're reading won't be available. I think this whole function should probably be moved into sky.catalog.common.